test/unit/deadline_client/cli/test_cli_asset.py

```diff
@@ -0,0 +1,157 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+import pytest
+from unittest.mock import patch, Mock
+from click.testing import CliRunner
+
+from deadline.client.cli import main
+from deadline.client import api
+from deadline.job_attachments.upload import S3AssetManager
+from deadline.job_attachments.models import AssetRootGroup
+
+
+@pytest.fixture
+def mock_prepare_paths_for_upload():
+    with patch.object(S3AssetManager, "prepare_paths_for_upload") as mock:
+        yield mock
+
+
+@pytest.fixture
+def mock_hash_attachments():
+    with patch.object(api, "hash_attachments", return_value=(Mock(), [])) as mock:
+        yield mock
+
+
+@pytest.fixture
+def asset_group_mock(tmp_path):
+    root_dir = str(tmp_path)
+    return AssetRootGroup(
+        root_path=root_dir,
+        inputs=set(),
+        outputs=set(),
+        references=set(),
+    )
+
+
+@pytest.fixture
+def upload_group_mock(asset_group_mock):
+    return Mock(
+        asset_groups=[asset_group_mock],
+        total_input_files=1,
+        total_input_bytes=100,
+    )
+
+
+class TestSnapshot:
+
+    def test_snapshot_root_directory_only(
+        self, tmp_path, mock_prepare_paths_for_upload, mock_hash_attachments, upload_group_mock
+    ):
+        """
+        Tests if CLI snapshot command calls correctly with an exiting directory path at --root-dir
+        """
+        root_dir = str(tmp_path)
+
+        temp_file = tmp_path / "temp_file.txt"
+        temp_file.touch()
```

Comment:
nit: Ideally we shouldn't be creating files/dirs in our unit tests, but we can address this later.